fixed wrong historical fact in Symbol.unscopables page (#34646)
mdn/content: files/en-us/web/javascript/reference/global_objects/array/@@unscopables/index.md

## Examples

Imagine the `values.push('something')` call below is in code that was written prior to ECMAScript 2015.

```js
var values = [];

with (values) {
  values.push("something");
}
```

> This code can be broken even if it does not modify the `values` array at all. This happens when the `values` property is added to `Array.prototype`. When the `values` method is added to the array as follows, the `with` statement in the above code interprets `values` as `Array.prototype.values` instead of the external variable.
> 
> ```js
> Array.prototype.values = function () {
>   // some code
> };
> ```
> 
> If the `@@unscopables` data property had not been introduced, the `values.push('something')` call would break because the JavaScript runtime would have interpreted `values` as {{jsxref("Array.prototype.values()")}} instead of the `values` array defined in the example code. This really happend when ECMAScript 2015 introduced the {{jsxref("Array.prototype.values()")}} method and caused a bug in Firefox ([Firefox Bug 883914](https://bugzilla.mozilla.org/show_bug.cgi?id=883914)).

Reviewer: ```suggestion
When ECMAScript 2015 introduced the {{jsxref("Array.prototype.values()")}} method, the `with` statement in the above code interprets `values` as the `values.values` property instead of the external `values` variable. If the `@@unscopables` data property had not been introduced, the `values.push('something')` call would break because it's now accessing `push` on the `values.values` method. This caused a bug to be reported to Firefox ([Firefox Bug 883914](https://bugzilla.mozilla.org/show_bug.cgi?id=883914)).
```

Author: I have made the changes according to your review. However, I have modified the "values.values property" to an "values.values array method" for the consistency. Is it ok?
